### PR TITLE
Use proxyMetadata in gateway env

### PR DESCRIPTION
**Pull request by howardjohn**
Avoid having to specify it twice like in
https://github.com/istio/istio/pull/24161#issuecomment-637939533

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by hovonako** [comment]
Thank @howardjohn

File: manifests/charts/gateways/istio-ingress/templates/deployment.yaml [hunk 1]
```diff
@@ -210,6 +210,10 @@ spec:
           - name: {{ $key }}
             value: {{ $val }}
           {{- end }}
+          {{- range $key, $value := .Values.meshConfig.defaultConfig.proxyMetadata }}
```
Comment (costinm) [inline manifests/charts/gateways/istio-ingress/templates/deployment.yaml:213]: Yes, in particular this was not correct - 'defaultConfig' is different from the 'actual config' for this deployment, i.e. the default is supposed to only provide defaults.

Comment (howardjohn) [reply manifests/charts/gateways/istio-ingress/templates/deployment.yaml:213]: Another option I thought of was reading proxy metadata within istio-agent and then dynamically setting the environment. But I think many of our environment variables need to be set when the process starts so not sure its any better

**Review by costinm — COMMENTED** [review]
To confirm: the proxyMetadata is still read somewhere in istio-agent ? 

I would like to make sure gateways are consistent with sidecars - for sidecars we allow the ProxyConfig annotation which is used, while gateways seem to lack this and use the 'default'. Probably not related with this change...

To avoid repeating past mistakes, it would be great if we could merge a user-specified configmap with an install-time ProxyConfig from values.yaml.

**Comment by howardjohn** [comment]
> To confirm: the proxyMetadata is still read somewhere in istio-agent ?
> 
> I would like to make sure gateways are consistent with sidecars - for sidecars we allow the ProxyConfig annotation which is used, while gateways seem to lack this and use the 'default'. Probably not related with this change...
> 
> To avoid repeating past mistakes, it would be great if we could merge a user-specified configmap with an install-time ProxyConfig from values.yaml.

Gateway properly overrides by the annotation like sidecars. I am not sure how to do that for proxyMetadata though (see comment above) - but would love to if we can

**Review by elfinhe — APPROVED** [review]
LGTM

**Comment by myidpt** [comment]
I added cherry-pick to 1.6 since we also want this in 1.6.

**Review by morvencao — APPROVED** [review]

**Comment by dgn** [comment]
I'm fine with doing this, but wouldn't the cleanest solution be to dynamically update these values based on MeshConfig? It becomes harder and harder to know which parts of MeshConfig apply immediately and which require a re-installation

**Pull request merged**